dbt/include/athena/macros/materializations/models/incremental/incremental.sql

```diff
@@ -81,6 +81,14 @@
     {% do run_query(create_tmp_table_iceberg(tmp_relation, sql, staging_location, false)) %}
     {% set build_sql = iceberg_merge(tmp_relation, target_relation, unique_key) %}
     {% do to_drop.append(tmp_relation) %}
+  {% else %}
+    {% set tmp_relation = make_temp_relation(target_relation) %}
+    {% if tmp_relation is not none %}
+      {% do adapter.drop_relation(tmp_relation) %}
+    {% endif %}
+    {% do run_query(create_table_as(True, tmp_relation, sql)) %}
+    {% set build_sql = incremental_insert(on_schema_change, tmp_relation, target_relation, existing_relation) %}
+    {% do to_drop.append(tmp_relation) %}
```

Comment:
Instead of adding this immediately after the partitioned_by getter:

```
{% if partitioned_by is none %}
   {% set raw_strategy = 'append' %}
{% endif %}
```

doing so we make it explicitly that the strategy in case of not partitions is append, and we keep the code more dry.